src/com/moclojer/specs/moclojer.clj

```diff
@@ -28,61 +40,98 @@
-    (try
+  (try
+    (let [{:keys [content error?]} (-> response
+                                       (get-response-body request)
+                                       (render-template request))]
       (cond
-        error? (let [parsed (json/read-str content :key-fn keyword)]
-                 {:error parsed
-                  :message (get parsed :message)})
-        (string? content) (try
-                            (json/read-str content)
-                            (catch Exception e
-                              (log/log :warn :json-parse-error
-                                       :body content
-                                       :message (.getMessage e))
-                              content))
-        :else content)
-      (catch Exception e
-        (log/log :error :bad-body
-                 :body content
-                 :message (.getMessage e))
-        {:error (.getMessage e)}))))
+        error?
+        (-> content
+            (json/read-str :key-fn keyword)
+            (as-> parsed {:error parsed
+                          :message (:message parsed)}))
+
+        (string? content)
+        (parse-json-safely content)
+
+        :else content))
+    (catch Exception e
+      (log/log :error :bad-body
+               :body (:body response)
+               :message (.getMessage e))
+      {:error (.getMessage e)})))
 
 (defn assoc-if [m k v]
   (if v
     (assoc m k v)
     m))
 
 (defn webhook-condition
-  "Checks `if` condition and return boolean value, default is true"
+  "Checks `if` condition and return boolean value, default is true when condition is empty"
   [condition request]
-  (let [template (str "{% if " condition " %}true{% else %}false{% endif %}")]
-    (if (empty? condition)
-      true
-      (boolean (Boolean/valueOf (selmer/render (ext-body/->str template) request))))))
+  (if (empty? condition)
+    true
+    (let [template (str "{% if " condition " %}true{% else %}false{% endif %}")]
+      (-> template
+          ext-body/->str
+          (selmer/render request)
+          Boolean/valueOf
+          boolean))))
 
 (defn build-parameters [request-values]
   (let [query (:query request-values)
         path (:path request-values)
         body (:body request-values)]
-    (-> (assoc-if {} :query-params query)
-        (assoc-if :path-params path)
-        (assoc-if :json-params body))))
+    (-> {}
+        (cond-> query (assoc :query-params query))
+        (cond-> path (assoc :path-params path))
+        (cond-> body (assoc :json-params body)))))
+
+(defn process-ws-message
+  "Process WebSocket message based on pattern matching"
+  [channel message pattern-configs request-params]
+  (let [params (assoc request-params :message message)]
+    (doseq [{:keys [pattern response]} pattern-configs]
+      (let [template (str "{% if message = \"" pattern "\" %}true{% else %}false{% endif %}")
+            matches (boolean (Boolean/valueOf (selmer/render template params)))]
```

Comment:
Why do you need to parse the pattern through selmer? Wouldn't a simple reg-expr work?

Reply:
This also opens possibilities for attacks.

Reply:
Simplicity of using selmer lol
fixed :P